docs/topics/coalesce-swashbuckle.md

```diff
@@ -0,0 +1,102 @@
+
+```xml:no-line-numbers{3}
+<ItemGroup>
+  <PackageReference Include="IntelliTect.Coalesce.Vue" Version="$(CoalesceVersion)" />
+  <PackageReference Include="IntelliTect.Coalesce.Swashbuckle" Version="$(CoalesceVersion)" />
+</ItemGroup>
+```
+
+### Configure Swagger in Program.cs
+
+Update your Program.cs file to configure Swagger and include Coalesce-specific enhancements. This involves [setting up Swagger as usual](https://learn.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?view=aspnetcore-8.0&tabs=visual-studio) and then applying the Coalesce configuration (Note: You do not need to install the `Swashbuckle.AspNetCore` package if you are using the coalesce one).
+
+```c#:no-line-numbers
+builder.Services.AddSwaggerGen(config =>
+{
+    config.SwaggerDoc("v1", new OpenApiInfo { Title = "My API", Version = "v1" });
+    config.AddCoalesce(); // Add coalesce specific configuration
+});
+```
+
+```c#
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+```
+
+## Improvements
+
+### Default Swagger Generation
+
+By default, Swagger in ASP.NET Core offers a simple way to document APIs. It generates API documentation based on the structure of your controller actions and data models. While this default setup is functional for many scenarios, it may fall short in representing more complex cases, especially when dealing with Coalesce-generated endpoints that include DataSources and Behaviors. These scenarios can lead to verbose and sometimes confusing Swagger documentation.
+
+### Coalesce Enhancements
+
+The `IntelliTect.Coalesce.Swashbuckle` package addresses the limitations of the default Swagger generation by providing custom Swagger filters. These filters enhance the readability and usability of your Swagger documentation for Coalesce-generated APIs.
+
+#### CoalesceDocumentFilter:
+
+**Purpose:** Cleans up unnecessary or redundant schema definitions in the Swagger document.
+
+**Functionality:** The filter removes empty or irrelevant definitions related to IDataSource and IBehaviors. These are interfaces used internally by Coalesce but do not need to appear in the Swagger documentation. By removing these definitions, the Swagger UI becomes more focused and less cluttered.
+
+#### CoalesceApiOperationFilter
+
+**Purpose:** Helps refine the operational details in the Swagger documentation for Coalesce-generated endpoints.
+
+**Functionality:** The filter processes various aspects of the API operations to provide cleaner generated API docs:
+
+- DTO Parameters: Adjusts the representation of DTO parameters to ensure only relevant properties are included.
+
+- Data Sources: Enhances the description and the schema of data source parameters so they are accurately represented.
+
+- Standard Parameters: It removes unnecessary parameters related to behaviors and data sources, improving the clarity of the generated swagger API documentation.
```

Comment:
I think just a simple high level summary of the effects is OK. We don't need to detail the exact functionality of each individual Filter implementation class - we can consider those as internal implementation details. The user will get a good enough idea of the effects.

```suggestion
The primary effect is an adjustment of parameter definitions to account for Coalesce's custom model binders that create instances of Data Sources and Behaviors on each request. These parameters will be updated in the OpenAPI document to account for data source parameters, filter parameters, and other model-specific customizations.
```